### PR TITLE
[WIP][Doc] Add  pipeline doc of rtmdet_description.md 

### PR DESCRIPTION
**Pull request by wanghonglie**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Motivation

Please describe the motivation of this PR and the goal you want to achieve through this PR.

## Modification

Please briefly describe what modification is made in this PR.

## BC-breaking (Optional)

Does the modification introduce changes that break the backward-compatibility of the downstream repos?
If so, please describe how it breaks the compatibility and how the downstream projects should modify their code to keep compatibility with this PR.

## Use cases (Optional)

If this PR introduces a new feature, it is better to list some use cases here, and update the documentation.

## Checklist

1. Pre-commit or other linting tools are used to fix the potential lint issues.
2. The modification is covered by complete unit tests. If not, please add more unit test to ensure the correctness.
3. If the modification has potential influence on downstream projects, this PR should be tested with downstream projects, like MMDetection or MMClassification.
4. The documentation has been modified accordingly, like docstring or example tutorials.

### CODE REVIEW

File: docs/zh_cn/algorithm_descriptions/rtmdet_description.md [hunk 1]
```diff
@@ -0,0 +1,76 @@
+## 3. 数据增强
+
+rtmdet采用了多种数据增强的方式来增加模型的性能，主要包括Mosaic，RandomResize，RandomCrop，YOLOXHSVRandomAug，RandomFlip和MixUP。
```
Comment (hhaAndroid) [inline docs/zh_cn/algorithm_descriptions/rtmdet_description.md:3]: RTMDet

Comment (hhaAndroid) [reply docs/zh_cn/algorithm_descriptions/rtmdet_description.md:3]: 中英文之间要空格

File: docs/zh_cn/algorithm_descriptions/rtmdet_description.md [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+
+训练阶段数据增强流程：
+![pipeline](https://user-images.githubusercontent.com/33799979/192729900-9d8a98b7-d9a3-4278-b87d-24f16258cfc8.png)
+
+### 3.1 Mosaic
```
Comment (hhaAndroid) [inline docs/zh_cn/algorithm_descriptions/rtmdet_description.md:8]: 一开始就要强调 cache 版本和 v5 版本的差异

**Comment by Nioolek** [comment]
英文的前后需要添加空格，以美化格式

File: docs/zh_cn/algorithm_descriptions/rtmdet_description.md [hunk 3]
```diff
@@ -0,0 +1,110 @@
+## 3. 数据增强
+
+RTMDet采用了多种数据增强的方式来增加模型的性能，主要包括单图数据增强:
+
+- **RandomResize 随机尺度变换**
+- **RandomCrop 随机裁剪**
+- **HSVRandomAug 颜色空间曾倩**
```
Comment (Nioolek) [inline docs/zh_cn/algorithm_descriptions/rtmdet_description.md:7]: 增强

File: docs/zh_cn/algorithm_descriptions/rtmdet_description.md [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+- **Mosaic 马赛克**
+- **MixUP 图像混合**
+
+MMDetection 开源库中已经对单图数据增强进行了封装，用户通过简单的修改配置即可使用库中提供的任何数据增强功能，且都是属于比较常规的数据增强，不需要特殊介绍。下面将具体介绍混合类数据增强的具体实现。
+
+与YOLOv5不同的是，RTMDet的s和tiny模型也采用了MixUP的数据增强，并且为混合类数据增强引入了Cache方案，有效地减少了图像处理的时间。具体介绍如下：
+
+### 3.1 为图像混合数据增强引入Cache
+
+Mosaic&Mixup 涉及到多张图片的混合，它们的耗时会是普通数据增强的K倍(K为混入图片的数量)。
```
Comment (Nioolek) [inline docs/zh_cn/algorithm_descriptions/rtmdet_description.md:21]: 这样写是否更好理解：
yolov5中，每次做mosaic时，4张图都需要现取。而在rtmdet中，主图需要现取，另外3张图来源于缓存cache。

File: docs/zh_cn/algorithm_descriptions/rtmdet_description.md [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+RTMDet将原本从数据集取图片改为从缓存队列种获取，通过牺牲一定空间的方式大幅提升了效率。
```
Comment (Nioolek) [inline docs/zh_cn/algorithm_descriptions/rtmdet_description.md:22]: 队列中

File: docs/zh_cn/algorithm_descriptions/rtmdet_description.md [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+else:
+    assert 'dataset' in results
+    # Be careful: deep copying can be very time-consuming
+    # if results includes dataset.
+    dataset = results.pop('dataset', None)
+```
+
+### 3.2 Mosaic
+
+mosaic 是将 4 张图拼接为 1 张大图，变相的相当于增加了 batch size，具体步骤为：
```
Comment (Nioolek) [inline docs/zh_cn/algorithm_descriptions/rtmdet_description.md:55]: 变相的相当于-〉相当于变相的

File: docs/zh_cn/algorithm_descriptions/rtmdet_description.md [hunk 7]
```diff
@@ -0,0 +1,110 @@
...
+- **RandomFlip 随机水平翻转**
+
+以及混合类数据增强：
+
+- **Mosaic 马赛克**
+- **MixUP 图像混合**
+
+MMDetection 开源库中已经对单图数据增强进行了封装，用户通过简单的修改配置即可使用库中提供的任何数据增强功能，且都是属于比较常规的数据增强，不需要特殊介绍。下面将具体介绍混合类数据增强的具体实现。
+
+与YOLOv5不同的是，RTMDet的s和tiny模型也采用了MixUP的数据增强，并且为混合类数据增强引入了Cache方案，有效地减少了图像处理的时间。具体介绍如下：
```
Comment (Nioolek) [inline docs/zh_cn/algorithm_descriptions/rtmdet_description.md:17]: 建议改成：
1.yolov5认为在s和nano模型上使用mixup是过剩的，小模型不需要这么强的数据增强。而rtmdet在s和tiny上也使用了mixup，并通过训练证明这个操作是有效的。
2.rtmdet在mixup和mosaic中引入cache机制。

File: docs/zh_cn/algorithm_descriptions/rtmdet_description.md [hunk 8]
```diff
@@ -0,0 +1,110 @@
+## 3. 数据增强
+
+RTMDet采用了多种数据增强的方式来增加模型的性能，主要包括单图数据增强:
```
Comment (hhaAndroid) [inline docs/zh_cn/algorithm_descriptions/rtmdet_description.md]: 中英文要间隔

File: docs/zh_cn/algorithm_descriptions/rtmdet_description.md (hunk 3 above)
Comment (hhaAndroid) [reply docs/zh_cn/algorithm_descriptions/rtmdet_description.md]: 可以绘制一个非常简单的流程图来表示整个数据的运行流，这样写用户不知道运行顺序

File: docs/zh_cn/algorithm_descriptions/rtmdet_description.md [hunk 9]
```diff
@@ -0,0 +1,108 @@
...
+
+MMDetection 开源库中已经对单图数据增强进行了封装，用户通过简单的修改配置即可使用库中提供的任何数据增强功能，且都是属于比较常规的数据增强，不需要特殊介绍。下面将具体介绍混合类数据增强的具体实现。
+
+与 YOLOv5 不同的是，YOLOv5认为在 s 和 nano 模型上使用 mixup 是过剩的，小模型不需要这么强的数据增强。而 RTMDet 在 s 和 tiny 上也使用了 MixUP，并通过训练证明这个操作是有效的。并且 RTMDet 为混合类数据增强引入了 Cache 方案，有效地减少了图像处理的时间。具体介绍如下：
+
+### 3.1 为图像混合数据增强引入Cache
+
+Mosaic&Mixup 涉及到多张图片的混合，它们的耗时会是普通数据增强的K倍(K为混入图片的数量)。 如在YOLOv5中，每次做 Mosaic 时， 4张图片的信息都需要从硬盘中重新加载。 而 RTMDet 只需要重新载入当前的一张图片，其余参与混合增强的图片则从缓存队列中获取，通过牺牲一定空间的方式大幅提升了效率。 另外，通过调整 cache 的大小以及 pop 的方式，也可以调整增强的强度。当使用较小的 cache 时，其效果类似 repeated augmentation。
+![cache](https://user-images.githubusercontent.com/33799979/192730011-90e2a28d-e163-4399-bf87-d3012007d8c3.png)
+如图所示，cache 队列中预先储存了N张已加载的图像与标签数据，每一个训练 step 中只需加载一张新的图片及其标签数据并更新到 cache 队列中，同时如果 cache 队列长度超过预设长度，则随机 pop 一张图（为了 tiny 模型训练更稳定，在 tiny 模型中不采用随机 pop 的方式），当需要进行混合数据增强时，只需要从cache中随机选择需要的图像进行拼接等处理，而不需要全部从硬盘中加载，节省了图像加载的时间。
```
Comment (hhaAndroid) [inline docs/zh_cn/algorithm_descriptions/rtmdet_description.md:23]: 在 tiny 模型中不采用随机 pop 的方式，而是移除最先加入的图片

File: docs/zh_cn/algorithm_descriptions/rtmdet_description.md [hunk 10]
```diff
@@ -0,0 +1,108 @@
...
+
+> cache 队列的最大长度 N 为可调整参数，Mosaic 增强默认 N=40， MixUP 所需图像数量相较 Mosaic 更少，因此 cache 数量默认为20， tiny 模型需要更稳定的训练条件，因此其 cache 数量也为其余规格模型的一半（ MixUP 为10，Mosaic 为20）
```
Comment (hhaAndroid) [inline docs/zh_cn/algorithm_descriptions/rtmdet_description.md:25]: 要说明这个 40 咋算出来的

File: docs/zh_cn/algorithm_descriptions/rtmdet_description.md [hunk 11]
```diff
@@ -0,0 +1,108 @@
...
+    img_i, (int(w_i * scale_ratio_i), int(h_i * scale_ratio_i)))
+```
+
+4. 拼接后，把 bbox 和 label 全部拼接起来，然后对 bbox 进行裁剪但是不过滤(可能出现一些无效框)
+
+```python
+mosaic_bboxes.clip_([2 * self.img_scale[0], 2 * self.img_scale[1]])
+```
+
+### 3.3 MixUP
```
Comment (hhaAndroid) [inline docs/zh_cn/algorithm_descriptions/rtmdet_description.md:98]: 可以简单说一下，实现上和 yolox 的 mixup 完全一样，只不过增加了类似 cache mosaic 的 cache 功能

File: docs/zh_cn/algorithm_descriptions/rtmdet_description.md [hunk 12]
```diff
@@ -0,0 +1,108 @@
+## 3. 数据增强
+
+RTMDet 采用了多种数据增强的方式来增加模型的性能，主要包括单图数据增强:
+
+- **RandomResize 随机尺度变换**
```
[the remark this answers is not in the document]
Comment (wanghonglie) [reply docs/zh_cn/algorithm_descriptions/rtmdet_description.md:5]: 这个放在最后那张图，需要提前放吗，还是加个超链接引用

Comment (hhaAndroid) [inline docs/zh_cn/algorithm_descriptions/rtmdet_description.md:5]: RandomResize 这个在 大模型 mLX 和 stn 上是不一样的，大模型由于参数较多，可以使用 large scale jitter 策略即参数为 (0.1,2.0)，而小模型采用 stand scale jitter 策略即 (0.5, 2.0) 策略

**Review by hhaAndroid — APPROVED** [review]

**Pull request merged**